iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -72,11 +88,61 @@ class ServiceRegistry
     const ServiceDescriptionVector_t getServices() const noexcept;
 
   private:
-    /// @todo #859 replace std::multimap with prefix tree
-    ::std::multimap<capro::IdString_t, uint64_t> m_serviceMap;
-    ::std::multimap<capro::IdString_t, uint64_t> m_instanceMap;
-    ::std::multimap<capro::IdString_t, uint64_t> m_eventMap;
-    ServiceDescriptionVector_t m_serviceDescriptionVector;
+    using Entry_t = cxx::optional<ServiceDescriptionEntry>;
+    using ServiceDescriptionContainer_t = cxx::vector<Entry_t, MAX_SERVICE_DESCRIPTIONS>;
+
+    static constexpr uint32_t NO_INDEX = MAX_SERVICE_DESCRIPTIONS;
+
+    // tag type for internal overloads
+    struct Wildcard
```

Comment:
Why not use the definition from: `iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp:33`?

This is compile time and has no overhead.

Reply:
Can do, we already have this dependency.